### PR TITLE
fix: produce JSONPatch patches with removals in the right order

### PR DESCRIPTION
**Pull request by fiam**
- Replace github.com/mattbaird/jsonpatch with github.com/wI2L/jsondiff
- Add support for live queries in the TS client
- Make subscriptions cancellable
- Enable jsonpatch for subscriptions over fetch
- Add integration test using jsonpatch over a live query

## Motivation and Context

This caused a problem every time a live query deleted multiple entries at the end
of the list at the same time.

## Marketing Changelog

Live Queries now work with bulk deletions

#### Checklist

- [ ] run `make test`
- [ ] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Code of conduct](https://github.com/wundergraph/wundergraph/blob/main/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by jensneuse** [comment]
Would it not make sense to fork the upstream, provide a fix, and use our fork until it's merged?

**Comment by fiam** [comment]
> Would it not make sense to fork the upstream, provide a fix, and use our fork until it's merged? I think someone mentioned it already: [mattbaird/jsonpatch#30](https://github.com/mattbaird/jsonpatch/issues/30)
> 
> So we can open a PR and help them improve their lib.

As discussed, we've added integration tests and replaced the library with another one.

File: packages/sdk/src/client/client.ts [hunk 1]
```diff
@@ -512,7 +516,18 @@ export class Client {
...
 		while (true) {
-			const { value, done } = await reader.read();
+			let value: Uint8Array | undefined;
+			let done: boolean;
+			try {
+				const result = await reader.read();
+				value = result.value;
+				done = result.done;
+			} catch (e) {
+				if (!(e instanceof Error) || !e.message.indexOf('aborted')) {
```
Comment (StarpTech) [inline packages/sdk/src/client/client.ts:526]: Could you add a comment for which case this is required? AbortController?

**Review by StarpTech — CHANGES_REQUESTED** [review]

**Review by StarpTech — APPROVED** [review]
LGTM

**Review by JivusAyrus — APPROVED** [review]

**Review by devsergiy — APPROVED** [review]

**Pull request merged**